bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleEngineCallbackImpl.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.automation.handler.RuleEngineCallback;
+
+/**
+ * @author Yordan Mihaylov - Initial Contribution
+ *
+ */
+public class RuleEngineCallbackImpl implements RuleEngineCallback {
+
+    private static Map<String, RuleEngineCallback> reCallbacks = new HashMap<String, RuleEngineCallback>();
```

Comment:
:)  of course I trust them, but I don't have full control over the removing objects. But it is ok for me to implement  reCallbacks as WeakHashMap. This is the easy and simple way.